dipy/workflows/reconst.py

```diff
@@ -21,6 +21,198 @@
 from dipy.workflows.workflow import Workflow
 
+from dipy.reconst import mapmri
+
+
+class ReconstMAPMRIFlow(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return 'mapmri'
+
+    def run(self, data_file, data_bvecs, data_bvals, small_delta, big_delta,
+            b0_threshold=0.0, laplacian=True, positivity=True,
+            bval_threshold=2000, save_metrics=[],
+            laplacian_weighting=0.05, radial_order=6, out_dir='',
+            out_rtop='rtop.nii.gz', out_lapnorm='lapnorm.nii.gz',
+            out_msd='msd.nii.gz', out_qiv='qiv.nii.gz',
+            out_rtap='rtap.nii.gz',
+            out_rtpp='rtpp.nii.gz', out_ng='ng.nii.gz',
+            out_perng='perng.nii.gz',
+            out_parng='parng.nii.gz'):
+        """ Workflow for fitting the MAPMRI model
+        (with optional Laplacian regularization).
+        Generates rtop, lapnorm, msd, qiv, rtap, rtpp,
+        non-gaussian (ng), parallel ng, perpendicular ng
+        saved in a nifti format in input files provided by
+        `data_file` and saves the nifti files to an output
+        directory specified by `out_dir`.
+
+        In order for the MAPMRI workflow to work in the way
+        intended either the laplacian or positivity or both must
+        be set to True.
+
+        Parameters
+        ----------
+        data_file : string
+            Path to the input volume.
+        data_bvecs : string
+            Path to the bvec files.
+        data_bvals : string
+            Path to the bval files.
+        small_delta : float
+            Small delta value used in generation of gradient table of provided
+            bval and bvec.
+        big_delta : float
+            Big delta value used in generation of gradient table of provided
+            bval and bvec.
+        b0_threshold : float, optional
+            Threshold used to find b=0 directions (default 0.0)
+        bval_threshold : float
+            Sets the b-value threshold to be used in the scale factor
+            estimation. In order for the estimated non-Gaussianity to have
+            meaning this value should set to a lower value (b<2000 s/mm^2)
+            such that the scale factors are estimated on signal points that
+            reasonably represent the spins at Gaussian diffusion.
+            (default: 2000)
+        save_metrics : list of strings
+            List of metrics to save.
+            Possible values: rtop, laplacian_signal, msd, qiv, rtap, rtpp,
+            ng, perng, parng
+            (default: [] (all))
+        laplacian_weighting : float
+            Weighting value used in fitting the MAPMRI model in the laplacian
+            and both model types. (default: 0.05)
+        radial_order : unsigned int
+            Even value used to set the order of the basis
+            (default: 6)
+        out_dir : string, optional
+            Output directory (default: input file directory)
+        laplacian : boolean
```

Comment:
Thanks for this last commit @aarya22!

`laplacian` and `positivity` should be before `out_dir`. can you move them? After that, it should be ready to go!

Reply:
@skoudoro Thanks for pointing that out. Should be good to go now.